builders.ncl

```diff
@@ -81,195 +73,158 @@ let { NickelDerivation, Derivation, .. } = import "contracts.ncl" in
           |> std.record.map
             (
               fun _n xs =>
-                # Intersperse ':' between environment variable fragments
-                let values = std.record.values xs in
-                let first = std.array.first values in
-                values
-                |> std.array.drop_first
-                |> std.array.fold_left (fun acc value => nix-s%"%{acc}:%{value}"%) first
+                # Intersperse ' ' between environment variable fragments
+                concatStringsSep " " (std.record.values xs)
             ),
-      } | (NickelPkg & { packages | { _ : Derivation } }),
+      } | NickelPkg,
+    },
+
+  Shell
+    | doc m%%%"
+      A derivation that is to be used as a shell, e.g. with `nix develop`.
+      Analogous to `mkShell`.
+    "%%%
+    = NixpkgsPkg & {
+      packages | doc "Packages to be added to the shell, setting PATH, LD_LIBRARY_PATH and other variables as needed" = {},
+      hooks | doc "Bash scripts to run when entering the shell" = {},
+
+      output = {
+        name | default = "shell",
+        version | default = "dev",
+
+        env.buildCommand = nix-s%"
+          echo "This derivation is not supposed to be built" 1>&2 1>/dev/null
+          exit 1
+        "%,
+        env.shellHook = concatStringsSep "\n" (std.record.values hooks),
+        structured_env.buildInputs = packages,
```

Comment:
Same thing about casing. Unless those are values that are then exported to Nix? Even there, I wonder if we shouldn't do the translation by hand between snake_case and camelCase. Otherwise it's going to be really confused as to which casing to use. Related discussion: https://github.com/tweag/nickel/discussions/1311

Reply:
This is passed to nixpkgs stdenv: https://nixos.org/manual/nixpkgs/stable/#var-stdenv-buildInputs .

Reply:
I see, we'll have to think about this at some point, but this PR might not be the right scope. If it's passed onto Nixpkgs, we don't have much choice, at least for now.